bindings/java/src/org/sleuthkit/datamodel/CommunicationsManager.java

```diff
@@ -0,0 +1,337 @@
+	
+	return db.getRelationshipTypes(account1.getArtifactId(), account2.getArtifactId());
+}
+
+/**
+ * Returns unique relation types between two accounts
+ * 
+ * @param account1 account 
+ * @param account2 account
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+public List <BlackboardArtifact> getRelationships(Account account1, Account account2) throws TskCoreException {
+	
+	return db.getRelationships(account1.getArtifactId(), account2.getArtifactId());
+}
+
+/**
+ * Returns unique relation types between two accounts
+ * 
+ * @param account1 account 
+ * @param account2 account
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+public List <BlackboardArtifact> getRelationshipsOfType(Account account1, Account account2, BlackboardArtifact.Type artifactType ) throws TskCoreException {
+	
+	return db.getRelationshipsOfType(account1.getArtifactId(), account2.getArtifactId(), artifactType);
+}
+
+/**
+ * Return folders found in the email source file
+ * 
+ * @param srcObjID pbjectID of the email PST/Mbox source file  
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+public List <MessageFolder> getMessageFolders(long srcObjID ) throws TskCoreException {
+	return db.getMessageFolders(srcObjID);
+}
+
+/**
+ * Return subfolders found in the email source file under the specified folder
+ * 
+ * @param srcObjID objectID of the email PST/Mbox source file  
+ * @param parentfolder parent folder  of messages to return
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+public List <MessageFolder> getMessageFolders(long srcObjID, MessageFolder  parentfolder ) throws TskCoreException {
+	return db.getMessageFolders(srcObjID, parentfolder);
+}
+
+/**
+ * Return email messages under given folder
+ * 
+ * @param parentfolder parent folder  of messages to return
+ * 
+ * @throws TskCoreException exception thrown if a critical error occurs
+ *                          within TSK core
+ */
+public List <BlackboardArtifact> getMessages(MessageFolder  parentfolder ) throws TskCoreException {
+	return db.getMessages(parentfolder);
+}
+
+/**
+ * Converts a list of accountIDs into a set of possible unordered pairs
+ * 
+ * @param accountIDs - list of accountID
+ * 
+ * @return Set<UnorderedPair<Long>>
+ */
+private Set<UnorderedPair<Long>> listToUnorderedPairs(List<Long> accountIDs) {
+	Set<UnorderedPair<Long>> relationships = new HashSet<UnorderedPair<Long>> ();
+	
+	for (int i = 0; i < accountIDs.size(); i++ ) {
+		for (int j = i+1; j < accountIDs.size(); j++ ) {
+			relationships.add(new UnorderedPair<Long> (accountIDs.get(i), accountIDs.get(j) ));
+		}
+	}
+	
+	return relationships;
+}
+
+public final class UnorderedPair<T> {
```

Comment:
Do we really need this class for such a lmited usecase?  Also it should probably be private.